[2023-12-02 16:29:57,095] {{taskinstance.py:670}} INFO - Dependencies all met for <TaskInstance: hw-3.4.load_data_psql 2023-12-02T15:52:55.287974+00:00 [queued]>
[2023-12-02 16:29:57,208] {{taskinstance.py:670}} INFO - Dependencies all met for <TaskInstance: hw-3.4.load_data_psql 2023-12-02T15:52:55.287974+00:00 [queued]>
[2023-12-02 16:29:57,210] {{taskinstance.py:880}} INFO - 
--------------------------------------------------------------------------------
[2023-12-02 16:29:57,212] {{taskinstance.py:881}} INFO - Starting attempt 7 of 7
[2023-12-02 16:29:57,213] {{taskinstance.py:882}} INFO - 
--------------------------------------------------------------------------------
[2023-12-02 16:29:57,243] {{taskinstance.py:901}} INFO - Executing <Task(PythonOperator): load_data_psql> on 2023-12-02T15:52:55.287974+00:00
[2023-12-02 16:29:57,252] {{standard_task_runner.py:54}} INFO - Started process 1592 to run task
[2023-12-02 16:29:57,355] {{standard_task_runner.py:77}} INFO - Running: ['airflow', 'run', 'hw-3.4', 'load_data_psql', '2023-12-02T15:52:55.287974+00:00', '--job_id', '16', '--pool', 'default_pool', '--raw', '-sd', 'DAGS_FOLDER/hw-3.4.py', '--cfg_path', '/tmp/tmpd5pqbbyc']
[2023-12-02 16:29:57,357] {{standard_task_runner.py:78}} INFO - Job 16: Subtask load_data_psql
[2023-12-02 16:29:57,432] {{logging_mixin.py:120}} INFO - Running <TaskInstance: hw-3.4.load_data_psql 2023-12-02T15:52:55.287974+00:00 [running]> on host 2cf373442eb7
[2023-12-02 16:29:57,478] {{base_hook.py:89}} INFO - Using connection to: id: conn1. Host: host.docker.internal, Port: 5434, Schema: quotes, Login: postgres, Password: XXXXXXXX, extra: None
[2023-12-02 16:29:58,002] {{logging_mixin.py:120}} INFO - ------------ response --------------
[2023-12-02 16:29:58,004] {{logging_mixin.py:120}} INFO - https://www.alphavantage.co/query?function=SYMBOL_SEARCH&apikey=PO8LZ4A0I0PPI60J&keywords=i
[2023-12-02 16:29:58,006] {{logging_mixin.py:120}} INFO - None
[2023-12-02 16:29:58,008] {{logging_mixin.py:120}} INFO - {'User-Agent': 'python-requests/2.23.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'}
[2023-12-02 16:29:58,011] {{logging_mixin.py:120}} INFO - ------------ end --------------
[2023-12-02 16:29:58,014] {{logging_mixin.py:120}} INFO - ------------ data --------------
[2023-12-02 16:29:58,017] {{logging_mixin.py:120}} INFO - {'bestMatches': [{'1. symbol': 'IX.LON', '2. name': 'I', '3. type': 'Equity', '4. region': 'United Kingdom', '5. marketOpen': '08:00', '6. marketClose': '16:30', '7. timezone': 'UTC+01', '8. currency': 'GBX', '9. matchScore': '1.0000'}, {'1. symbol': 'I01.FRK', '2. name': 'IsoEnergy Ltd', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I02.FRK', '2. name': 'IMAC HOLDINGS DL-001', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I0R.FRK', '2. name': 'FIRST AMERN URANIUM', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I10.FRK', '2. name': 'Italgas S.p.A', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I1D.FRK', '2. name': 'ARRAIL GROUP LTD.', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I1F.FRK', '2. name': 'Q4 INC.', '3. type': 'Equity', '4. region': 'Frankfurt', '5. marketOpen': '08:00', '6. marketClose': '20:00', '7. timezone': 'UTC+02', '8. currency': 'EUR', '9. matchScore': '0.3333'}, {'1. symbol': 'I1AC34.SAO', '2. name': 'IAC/InterActiveCorp', '3. type': 'Equity', '4. region': 'Brazil/Sao Paolo', '5. marketOpen': '10:00', '6. marketClose': '17:30', '7. timezone': 'UTC-03', '8. currency': 'BRL', '9. matchScore': '0.2000'}, {'1. symbol': 'I1CE34.SAO', '2. name': 'Intercontinental Exchange Inc.', '3. type': 'Equity', '4. region': 'Brazil/Sao Paolo', '5. marketOpen': '10:00', '6. marketClose': '17:30', '7. timezone': 'UTC-03', '8. currency': 'BRL', '9. matchScore': '0.2000'}, {'1. symbol': 'I1FO34.SAO', '2. name': 'Infosys Limited', '3. type': 'Equity', '4. region': 'Brazil/Sao Paolo', '5. marketOpen': '10:00', '6. marketClose': '17:30', '7. timezone': 'UTC-03', '8. currency': 'BRL', '9. matchScore': '0.2000'}]}
[2023-12-02 16:29:58,021] {{logging_mixin.py:120}} INFO - ------------ end --------------
[2023-12-02 16:29:58,023] {{taskinstance.py:1150}} ERROR - Ошибка записи данных: 'timestamp'
Traceback (most recent call last):
  File "/usr/local/airflow/dags/modules/functions.py", line 70, in load_data_psql
    get_date = datetime.utcfromtimestamp(data["timestamp"]).strftime('%Y-%m-%d %H:%M:%S')
KeyError: 'timestamp'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 984, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 113, in execute
    return_value = self.execute_callable()
  File "/usr/local/lib/python3.7/site-packages/airflow/operators/python_operator.py", line 118, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/usr/local/airflow/dags/modules/functions.py", line 83, in load_data_psql
    raise Exception(f'Ошибка записи данных: {error}')
Exception: Ошибка записи данных: 'timestamp'
[2023-12-02 16:29:58,032] {{taskinstance.py:1194}} INFO - Marking task as FAILED. dag_id=hw-3.4, task_id=load_data_psql, execution_date=20231202T155255, start_date=20231202T162957, end_date=20231202T162958
[2023-12-02 16:30:02,063] {{local_task_job.py:102}} INFO - Task exited with return code 1
